modules/core/src/test/java/org/apache/ignite/internal/GridJobWithSubTasksTest.java

```diff
@@ -0,0 +1,170 @@
+package org.apache.ignite.internal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCompute;
+import org.apache.ignite.cluster.ClusterNode;
+import org.apache.ignite.compute.ComputeJob;
+import org.apache.ignite.compute.ComputeJobAdapter;
+import org.apache.ignite.compute.ComputeJobResult;
+import org.apache.ignite.compute.ComputeTaskAdapter;
+import org.apache.ignite.compute.ComputeTaskFuture;
+import org.apache.ignite.compute.ComputeTaskMapAsync;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.util.future.IgniteFinishedFutureImpl;
+import org.apache.ignite.lang.IgniteFuture;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * This test can't end without IgniteCompute#withExecutore() because tasks will borrow all threads from the thread pool
+ * and will hang in waiting of subtasks, while subtasks are waiting for free threads.
+ */
+public class GridJobWithSubTasksTest extends GridCommonAbstractTest {
```

Comment:
may be add a specific test scenario. Which leads to deadlock in case of using regular executors.